platform/architecture-overview.svg

```diff
@@ -0,0 +1 @@
,2.642 0.354,3.957c0.14,1.173 0.313,2.343 0.529,3.504c0.172,0.922 0.37,1.839 0.605,2.747c0.152,0.589 0.319,1.174 0.507,1.752c0.139,0.428 0.289,0.852 0.453,1.271c0.126,0.319 0.259,0.635 0.402,0.947c0.105,0.228 0.216,0.454 0.333,0.677c0.076,0.145 0.154,0.288 0.236,0.429c0.046,0.08 0.092,0.159 0.14,0.237Z" style="fill:#fff;"/><path d="M466.82,405.738l-6.553,0l0,-0.003c-2.892,-0.197 -5.221,-9.21 -5.221,-20.299c0,-11.09 2.329,-20.103 5.221,-20.299l0,-0.004l6.553,0c-0.019,0.032 -0.038,0.063 -0.057,0.094c-0.084,0.142 -0.165,0.285 -0.244,0.43c-0.104,0.194 -0.204,0.39 -0.3,0.589c-0.122,0.255 -0.237,0.514 -0.347,0.775c-0.148,0.352 -0.284,0.709 -0.412,1.07c-0.176,0.497 -0.335,1 -0.481,1.507c-0.193,0.672 -0.363,1.351 -0.516,2.033c-0.238,1.064 -0.434,2.137 -0.598,3.214c-0.179,1.177 -0.32,2.359 -0.43,3.544c-0.124,1.316 -0.208,2.636 -0.259,3.957c-0.056,1.437 -0.072,2.875 -0.05,4.313c0.018,1.187 0.062,2.373 0.134,3.558c0.081,1.322 0.197,2.642 0.354,3.957c0.141,1.173 0.314,2.343 0.53,3.504c0.171,0.922 0.369,1.839 0.604,2.747c0.152,0.589 0.32,1.174 0.508,1.752c0.139,0.428 0.289,0.852 0.453,1.271c0.125,0.319 0.259,0.635 0.402,0.947c0.105,0.228 0.215,0.454 0.332,0.677c0.076,0.145 0.155,0.288 0.236,0.429c0.046,0.08 0.093,0.159 0.141,0.237Z" style="fill:#fff;"/></g></g></g><g><rect x="0" y="116.015" width="326.041" height="220.308"/><text x="132.06px" y="318.923px" style="font-family:'ArialMT', 'Arial', sans-serif;font-size:13.922px;fill:#e6e6e6;">Endpoints</text><g><path d="M176.954,249.139l0,11.166l-13.471,0l0,15.77l13.471,0l0,11.176l-38.113,0l0,-38.112l38.113,0Z" style="fill:#fff;fill-rule:nonzero;"/><rect x="165.783" y="262.61" width="22.342" height="11.165" style="fill:#fff;fill-rule:nonzero;"/></g><g><path d="M82.842,249.139l0,11.166l-13.471,0l0,15.77l13.471,0l0,11.176l-38.112,0l0,-38.112l38.112,0Z" style="fill:#fff;fill-rule:nonzero;"/><rect x="71.671" y="262.61" width="22.342" height="11.165" style="fill:#fff;fill-rule:nonzero;"/></g><g><path d="M271.065,249.139l0,11.166l-13.471,0l0,15.77l13.471,0l0,11.176l-38.112,0l0,-38.112l38.112,0Z" style="fill:#fff;fill-rule:nonzero;"/><rect x="259.894" y="262.61" width="22.342" height="11.165" style="fill:#fff;fill-rule:nonzero;"/></g><g id="NSB-wb"><rect x="50.872" y="142.271" width="53.96" height="53.96" style="fill:#00729c;"/><path d="M97.374,151.992l-12.084,24.067c-0.598,0.874 -1.83,1.03 -2.625,0.304l-10.274,-10.274l-10.962,21.848l-3.014,-1.512c0,0 12.084,-24.068 12.084,-24.068c0.598,-0.874 1.83,-1.029 2.625,-0.303l10.274,10.273l10.962,-21.847c1.005,0.504 2.009,1.008 3.014,1.512Z" style="fill:#fff;fill-rule:nonzero;"/><rect x="56.877" y="148.275" width="6.005" height="6.005" style="fill:#00a3c4;"/><rect x="68.887" y="148.275" width="6.005" height="6.005" style="fill:#00a3c4;"/><rect x="80.897" y="148.275" width="6.005" height="6.005" style="fill:#00a3c4;"/><rect x="68.887" y="160.201" width="6.005" height="6.005" style="fill:#00a3c4;"/><rect x="80.897" y="160.201" width="6.005" height="6.005" style="fill:#00a3c4;"/><
```

Comment:
This digram looks much better than the old one. There's just one thing I find strange: the arrows are shown going _from_ the queues _to_ Service Control, which seems like the wrong direction. In the old diagram there was an arrow head going _from_ the error queue _to_ ServiceControl, which seems correct. Should the arrows be reversed?

Reply:
Same thing with SP and SI (although the direction does match the old diagram). I realise that there is a level of communication on both directions, but it feels like the main flow of information should be shown in the other direction. I.e. messages go _from_ endpoints _to_ the queues, and then _from_ the queues _to_ ServiceControl. Then that information goes _from_ ServiceControl _to_ SP and SI.

Reply:
There's some room for interpretation here. In one sense, NSB pushes messages to the queues and SC pulls messages from them. Similarly, SP and SI pull data from SC. So maybe that's the rational for the arrow heads, not so much that the data flows in a certain direction but that's how each piece interacts to push/pull the information.

That said, mapping the direction the data flows probably makes more sense for an architectural diagram.

@sergioc Thoughts?

Reply:
I think the push/pull aspect is less interesting than the data flow. After all, push/pull is just an implementation detail. E.g. you could potentially switch to a different API were SC pushes data to SP.

Reply:
I agree with @adamralph. I'll update the diagram.